src/traces/scatter/link_traces.js

```diff
@@ -0,0 +1,43 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Plots = require('../../plots/plots');
+
+module.exports = function linkTraces(gd, plotinfo, cdscatter) {
+    var i, cd, trace;
+    var xa = plotinfo.x();
+    var ya = plotinfo.y();
+
+    var prevtrace = null;
+
+    for(i = 0; i < cdscatter.length; ++i) {
+        cd = cdscatter[i];
+        trace = cd[0].trace;
+
+        if(trace.visible === true && (
+            (Plots.traceIs(trace, 'cartesian') && trace.xaxis === xa._id && trace.yaxis === ya._id) ||
+            trace.type === 'scatterternary'
```

Comment:
Yeah, I'd support thinking through this and paring it down to *only* checks that actually have any meaningful effect.